mk/opencl.mk

```diff
@@ -1,7 +1,13 @@
-# FIXME: perform sanity checks before compilation
-OPENCL_LIB ?= /usr/local/cuda-9.1/lib64
+OPENCL_LIB_PATH ?= /usr/local/cuda/lib64
+
+OPENCL_LIB := $(shell test -d $(OPENCL_LIB_PATH); echo $$?)
```

Comment:
Can you validate some critical headers as well?

Reply:
Which headers do you refer? Something like `CL/cl.h` in this case?

Reply:
Users might specify other OpenCL implementation rather than the one shipped by NVIDIA SDK.  Take [Beignet](https://www.freedesktop.org/wiki/Software/Beignet/) for example, it is pure software implementation and suitable for CI purpose. However, it is hard for current scripts to detect its availability.

Reply:
I would give it a try.

Reply:
The availability of Beignet should be included in another issue #69. The original goal of this PR is achieved.